omnigibson/prims/entity_prim.py

```diff
@@ -955,76 +956,114 @@ def get_relative_linear_velocity(self):
         Returns:
             3-array: (x,y,z) Linear velocity of root link in its own frame
         """
-        return T.quat2mat(self.get_orientation()).T @ self.get_linear_velocity()
+        return T.quat2mat(self.get_position_orientation()[1]).T @ self.get_linear_velocity()
 
     def get_relative_angular_velocity(self):
         """
         Returns:
             3-array: (ax,ay,az) angular velocity of root link in its own frame
         """
-        return T.quat2mat(self.get_orientation()).T @ self.get_angular_velocity()
+        return T.quat2mat(self.get_position_orientation()[1]).T @ self.get_angular_velocity()
+
+    def set_position_orientation(
+        self, position=None, orientation=None, frame: Literal["world", "parent", "scene"] = "world"
+    ):
+        """
+        Set the position and orientation of entry prim object.
+
+        Args:
+            position (None or 3-array): The position to set the object to. If None, the position is not changed.
+            orientation (None or 4-array): The orientation to set the object to. If None, the orientation is not changed.
+            frame (Literal): The frame in which to set the position and orientation. Defaults to world. parent frame
+            set position relative to the object parent. scene frame set position relative to the scene.
+        """
+
+        assert frame in ["world", "parent", "scene"], f"Invalid frame '{frame}'. Must be 'world', 'parent', or 'scene'."
+
+        # If we are in a scene, compute the scene-local transform before setting the pose
+        if frame == "scene":
+            position, orientation = T.compute_scene_transform(self, position, orientation, frame)
 
```

Comment:
Again, very unclean code here. You convert to world here if its scene, then in the below lines you call the parent with converted coordinates but the scene frame. So there is a huge error here and you don't notice it because you have spread out the conversions for the different frames. In these functions, if you are calling the parent directly, then you dont need to convert frames. But anytime you need to convert, make sure you convert BOTH scene AND local to world, and if you are calling any parent fns with the converted stuff, make sure you call with frame=world and not frame=frame